lib/convert.js

```diff
@@ -1,60 +1,106 @@
+ * @param {Boolean} [options.separateTextures=false] Write out separate textures only.
+ * @param {Boolean} [options.compress=false] Quantize positions, compress texture coordinates, and oct-encode normals.
+ * @param {Boolean} [options.optimize=false] Use the optimization stages in the glTF pipeline.
+ * @param {Boolean} [options.optimizeForCesium=false] Optimize the glTF for Cesium by using the sun as a default light source.
+ * @param {Boolean} [options.generateNormals=false] Generate normals if they are missing.
+ * @param {Boolean} [options.ao=false] Apply ambient occlusion to the converted model.
+ * @param {Boolean} [options.textureCompressionOptions] Options sent to the compressTextures stage of gltf-pipeline.
+ * @param {Boolean} [options.bypassPipeline=false] Bypass the gltf-pipeline for debugging purposes. This option overrides many of the options above and will save the glTF with the KHR_materials_common extension.
+ * @param {Boolean} [options.checkTextureAlpha=false] Do a more exhaustive check for texture transparency by looking at the alpha channel of each pixel.
+ */
+
+function convert(objPath, gltfPath, options) {
+    options = defaultValue(options, defaultValue.EMPTY_OBJECT);
     var binary = defaultValue(options.binary, false);
-    var embed = defaultValue(options.embed, true);
-    var embedImage = defaultValue(options.embedImage, true);
+    var separate = defaultValue(options.separate, false);
+    var separateTextures = defaultValue(options.separateTextures, false) || separate;
     var compress = defaultValue(options.compress, false);
-    var ao = defaultValue(options.ao, false);
+    var optimize = defaultValue(options.optimize, false);
     var optimizeForCesium = defaultValue(options.optimizeForCesium, false);
+    var generateNormals = defaultValue(options.generateNormals, false);
+    var ao = defaultValue(options.ao, false);
+    var textureCompressionOptions = options.textureCompressionOptions;
+    var bypassPipeline = defaultValue(options.bypassPipeline, false);
 
-    if (!defined(objFile)) {
-        throw new Error('objFile is required');
+    if (!defined(objPath)) {
+        throw new DeveloperError('objPath is required');
     }
 
-    if (!defined(outputPath)) {
-        outputPath = path.dirname(objFile);
+    if (!defined(gltfPath)) {
+        throw new DeveloperError('gltfPath is required');
     }
 
-    var inputPath = path.dirname(objFile);
-    var modelName = path.basename(objFile, '.obj');
-
-    var extension = path.extname(outputPath);
-    if (extension !== '') {
-        modelName = path.basename(outputPath, extension);
-        outputPath = path.dirname(outputPath);
+    var basePath = path.dirname(objPath);
+    var modelName = path.basename(objPath, path.extname(objPath));
+    var extension = path.extname(gltfPath);
+    if (extension === '.glb') {
```

Comment:
Even though `bypassPipeline` is supposed to be a 'only if you know what you're doing' option, should we have a check here if bypassing is happening too?